### DISCUSSION TITLE
Continuous return Bad_Timeout in SubscriptionListener.onPublishFailure

### DISCUSSION THREAD

**Discussion by fonlin** [eclipse-milo/milo#1063] [General]
The SubscriptionListener.onPublishFailure continuous report exception `Bad_Timeout`. However, `Subscription` looks normal and client can receive changed data at the same time.

I guess it has something to do with `PublishRequest Timeout Hint`. 

The sdk-client version is `0.6.8`.

![image](https://user-images.githubusercontent.com/31064021/195315462-408f789f-33ee-464f-92c1-76550fec4c27.png)

![image](https://user-images.githubusercontent.com/31064021/195315679-8eeb8709-2075-42c2-b3c6-01816c1f6ab0.png)

**Comment by fonlin** [comment]
I found a issue #679 , but i has checked Kepware and my client timestamp, and they are similar

**Comment by kevinherron** [comment]
Can you get a Wireshark capture?

This is possibly an issue with the server you're connected to and how it's handling receiving multiple PublishRequests. Some are timing out, but some are not, which is why your subscription continues to function.

I would imagine this is reproducible given the number of subscriptions you have created and the timing properties of each, so those details would be relevant too. So would the vendor and version of the server you are connected to.

**Comment by fonlin** [comment]
> Can you get a Wireshark capture?
> 
> This is possibly an issue with the server you're connected to and how it's handling receiving multiple PublishRequests. Some are timing out, but some are not, which is why your subscription continues to function.
> 
> I would imagine this is reproducible given the number of subscriptions you have created and the timing properties of each, so those details would be relevant too. So would the vendor and version of the server you are connected to.

OK, We have upgraded the Kepware version and will continue to observe this issue. Thanks a lot for your answer!
